### PR TITLE
Initial implemtation of rotation snap

### LINKED ISSUE #714: Snap to horizontal
Project are a little bit askew. Users struggle to neatly place images and text. Snap to horizontal (+/-5º) will make this much easier.

### PR DESCRIPTION
**Pull request by alanmoo**
May close #714, I'd like to get this into the beta so we can discuss how it feels.

### CODE REVIEW

**Comment by alanmoo** [comment]
I know the code's a little WET but I'm thinking this might need to be modified, so I wanted to leave the hooks in place for that.

File: src/components/basic-element/basic-element.jsx [hunk 1]
```diff
@@ -268,12 +268,30 @@ var BasicElement = React.createClass({
   },
 
   handleRotationAndScale: function(angle, scale) {
-    this.setState({
-      angle: angle,
```
Comment (gvn) [inline src/components/basic-element/basic-element.jsx:272]: This patch could be done a lot more concisely by just modifying this line to:

``` javascript
// Snap rotation
angle: Math.abs(angle) < 0.15 ? 0 : angle,
```

Comment (alanmoo) [inline src/components/basic-element/basic-element.jsx:272]: You make a great point. Hang on.

**Comment by gvn** [comment]
Feels pretty good to me! :+1:

**Pull request merged**